aws/logs_monitoring/template.yaml

```diff
@@ -391,6 +398,10 @@ Conditions:
     Fn::Equals:
       - Ref: DdForwardLog
       - false
+  SetDdStepFunctionTraceEnabled:
+    Fn::Equals:
+      - Ref: DdStepFunctionTraceEnabled
+      - true
```

Comment:
What will be the default value in CloudFormation console (shown as below) when customer install a new DD forwarder?
It should be default false, but customers can set it to be true if they want.

<img width="1375" alt="Screenshot 2024-07-31 at 11 27 05 AM" src="https://github.com/user-attachments/assets/5182e9f8-8bd3-4b35-80d9-bbc79bd5264c">

Reply:
The console will use the default value in `Parameters:` section, so in this case it would be `false`.

https://github.com/DataDog/datadog-serverless-functions/pull/831/files#diff-7c5bda646bf6435d9285913888d0473f0ac98ed0fdd025ae5d9ea357ad9b3ffaR164-R166

Reply:
This should be the default behavior, here is the flow

* Template has the default for the `DdStepFunctionTraceEnabled` to be `false`
* `SetDdStepFunctionTraceEnabled` condition will then be `false` because `DdStepFunctionTraceEnabled != true`
* `DD_STEP_FUNCTION_TRACE_ENABLED` will be `AWS::NoValue`, which means the env var won't be set